mathics/builtin/system.py

```diff
@@ -223,18 +224,13 @@ class Packages(Predefined):
     </dl>
 
     X> $Packages
-    = {CombinatoricaOld,ImportExport,Internal,System,XML}
-    #> MemberQ[$Packages, "System"]
+    = {CombinatoricaOld`,ImportExport`,XML`,Internal`,System`,Global`}
+    #> MemberQ[$Packages, "System`"]
     = True
     """
 
     name = "$Packages"
-
-    def evaluate(self, evaluation):
-        return Expression(
-            "List",
-            *(String(name) for name in evaluation.definitions.get_package_names()),
-        )
+    rules = {'$Packages': '{"CombinatoricaOld`", "ImportExport`",  "XML`","Internal`", "System`", "Global`"}',}
```

Comment:
I thought we got rid of `CombinatoricaOld`. I don't see it in the repository any more. 

The long-term plan was to be able to `Import[]` this directly from the authors' site. 

And for _testing_, I've been thinking of adding parts of it for like we do for the `GS.m`

Reply:
There is still the Builtin ```CombinatorycaOld`BinarySearch``` that is under this context. Shall we move it to ```System` ```?

Reply:
That's a possibility. The downside though is that this appears in under the `Combinatorica` namespace in [WL](https://reference.wolfram.com/language/Combinatorica/ref/BinarySearch.html). 

Another possibility is to get this working as a real package written in WL. 

In the more-examples branch I tried adding the [current Combinatorica BInarySearch](https://github.com/mathics/Mathics/blob/more-examples/examples/combinatorica/c-small.m#L170-L190). However doesn't seem to pass tests. 

If it is an easy thing to fix up the code (and it might be, but I don't know WL that well), we could add this as a True WL package instead of having this as `combinatorica/c-small.m`.  

I leave the choice of which way to go up to you.